src/ImageSharp/MetaData/Profiles/ICC/IccDataReader.cs

```diff
@@ -0,0 +1,1727 @@
+        /// </summary>
+        /// <param name="data">The data to read</param>
+        public IccDataReader(byte[] data)
+        {
+            this.data = data;
+        }
+
+        /// <summary>
+        /// Sets the reading position to the given value
+        /// </summary>
+        /// <param name="index">The new index position</param>
+        public void SetIndex(int index)
+        {
+            this.index = index.Clamp(0, this.data.Length);
+        }
+
+        #region Read Primitives
+
+        /// <summary>
+        /// Reads an ushort
+        /// </summary>
+        /// <returns>the value</returns>
+        public ushort ReadUInt16()
+        {
+            return this.converter.ToUInt16(this.data, this.AddIndex(2));
+        }
+
+        /// <summary>
+        /// Reads a short
+        /// </summary>
+        /// <returns>the value</returns>
+        public short ReadInt16()
+        {
+            return this.converter.ToInt16(this.data, this.AddIndex(2));
+        }
+
+        /// <summary>
+        /// Reads an uint
+        /// </summary>
+        /// <returns>the value</returns>
+        public uint ReadUInt32()
+        {
+            return this.converter.ToUInt32(this.data, this.AddIndex(4));
+        }
+
+        /// <summary>
+        /// Reads an int
+        /// </summary>
+        /// <returns>the value</returns>
+        public int ReadInt32()
+        {
+            return this.converter.ToInt32(this.data, this.AddIndex(4));
+        }
+
+        /// <summary>
+        /// Reads an ulong
+        /// </summary>
+        /// <returns>the value</returns>
+        public ulong ReadUInt64()
+        {
+            return this.converter.ToUInt64(this.data, this.AddIndex(8));
+        }
+
+        /// <summary>
+        /// Reads a long
+        /// </summary>
+        /// <returns>the value</returns>
+        public long ReadInt64()
+        {
+            return this.converter.ToInt64(this.data, this.AddIndex(8));
+        }
+
+        /// <summary>
+        /// Reads a float
+        /// </summary>
+        /// <returns>the value</returns>
+        public float ReadSingle()
+        {
+            return this.converter.ToSingle(this.data, this.AddIndex(4));
+        }
+
+        /// <summary>
+        /// Reads a double
+        /// </summary>
+        /// <returns>the value</returns>
+        public double ReadDouble()
+        {
+            return this.converter.ToDouble(this.data, this.AddIndex(8));
+        }
+
+        /// <summary>
+        /// Reads an ASCII encoded string
+        /// </summary>
+        /// <param name="length">number of bytes to read</param>
+        /// <returns>The value as a string</returns>
+        public string ReadASCIIString(int length)
+        {
+            // Encoding.ASCII is missing in netstandard1.1, use UTF8 instead because it's compatible with ASCII
+            string value = Encoding.UTF8.GetString(this.data, this.AddIndex(length), length);
```

Comment:
You could to `Encoding.GetEncoding("ASCII")` instead: https://github.com/JimBobSquarePants/ImageSharp/blob/a82d28f563279c433d9474981774c1745f8d25ca/src/ImageSharp/Formats/Png/PngDecoderOptions.cs#L15